### ISSUE TITLE
The loss function for each equation

### ISSUE THREAD

**Issue by KirillZubov** [SciML/NeuralPDE.jl#320]
For now, we create one loss function for all equations and one function for all boundary conditions.
The goal of the problem is to create loss functions for each equation and boundary conditions to provide more access to each loss function for a developer.

**Comment by ChrisRackauckas** [comment]
That would be useful for multi-objective optimization. Are there good optimizers that support that?

**Comment by KirillZubov** [comment]
Actually, this is a special case of what we call adaptive loss reweighting: `L(θ) = sum(w1(θ)*L1(θ),...,wn(θ)*Ln(θ)) `

The first thing I would like to do is update normalize coefficient(N) for the quadrature method.
`1/N*loss(θ)`
 we use N as the number of points and in the quadrature method, we do not know how many points, so there is now a very rough approximation `N = 10^dim `

I would like to try something like   `1 / @nograd f(quadrLoss(θ)) * quadrLoss(θ)`


Second is for system of pde,   `eq = f(u1,...,un)`
the gradient update needs to be balanced for each varibles,
`wi = optim(grad(u{n}))`
`L  = w1*L1 +.. +wn*Ln `
 but I guess @zoemcc seems to have moved a lot further than me here.


And last, interesting moment for me ​is the estimate between `grad(D(NN)) and  grad(NN))`
​D -derivative,  NN- neural network

we also use finite differences with simple two-point estimation and the error of derivative increases rapidly relate to the degree of derivative.
`error(NN)<error(D^1(NN))<<error(D^2(NN)) ...<<error(D^n(NN))`
